core/test/financial_templates/ExpiringMultiPartyCreator.js

```diff
@@ -0,0 +1,98 @@
+const Registry = artifacts.require("Registry");
+const ExpiringMultiParty = artifacts.require("ExpiringMultiParty");
+
+ERC20Mintable.setProvider(web3.currentProvider);
+
+contract("ExpiringMultiParty", function(accounts) {
+  let contractCreator = accounts[0];
+
+  // Contract variables
+  let collateralToken;
+  let finder;
+  let expiringMultiPartyCreator;
+  let registry;
+
+  // Re-used variables
+  let constructorParams;
+
+  beforeEach(async () => {
+    collateralToken = await ERC20Mintable.new("COLLATERAL_TOKEN", "COL", 18, { from: contractCreator });
+    finder = await Finder.deployed();
+    registry = await Registry.deployed();
+
+    expiringMultiPartyCreator = await ExpiringMultiPartyCreator.new(true, finder.address, { from: contractCreator });
+
+    await registry.addMember(RegistryRolesEnum.DERIVATIVE_CREATOR, expiringMultiPartyCreator.address, {
+      from: contractCreator
+    });
+
+    constructorParams = {
+      expirationTimestamp: "1234567890",
+      withdrawalLiveness: "1000",
+      collateralAddress: collateralToken.address,
+      tokenFactoryAddress: TokenFactory.address,
+      priceFeedIdentifier: web3.utils.utf8ToHex("UMATEST"),
+      syntheticName: "Test UMA Token",
+      syntheticSymbol: "UMATEST",
+      liquidationLiveness: "1000",
+      collateralRequirement: { rawValue: toWei("1.5") },
+      disputeBondPct: { rawValue: toWei("0.1") },
+      sponsorDisputeRewardPct: { rawValue: toWei("0.1") },
+      disputerDisputeRewardPct: { rawValue: toWei("0.1") }
+    };
+  });
+
+  it("Can create new instances of ExpiringMultiParty", async function() {
+    // Use `.call` to get the returned value from the function.
+    let functionReturnedAddress = await expiringMultiPartyCreator.createExpiringMultiParty.call(constructorParams, {
+      from: contractCreator
+    });
+
+    // Execute without the `.call` to perform state change. catch the result to query the event.
+    let createdAddressResult = await expiringMultiPartyCreator.createExpiringMultiParty(constructorParams, {
+      from: contractCreator
+    });
+
+    // Catch the address of the new contract from the event. Ensure that the assigned party member is correct.
+    let expiringMultiPartyAddress;
+    truffleAssert.eventEmitted(createdAddressResult, "CreatedExpiringMultiParty", ev => {
+      expiringMultiPartyAddress = ev.expiringMultiPartyAddress;
+      return ev.expiringMultiPartyAddress != 0 && ev.partyMemberAddress == contractCreator;
+    });
+
+    // Ensure value returned from the event is the same as returned from the function
+    assert.equal(functionReturnedAddress, expiringMultiPartyAddress);
+
+    // Instantiate an instance of the expiringMultiParty and check a few constants that should hold true
+    let expiringMultiParty = await ExpiringMultiParty.at(expiringMultiPartyAddress);
+    assert(await expiringMultiParty.expirationTimestamp(), constructorParams.expirationTimestamp);
```

Comment:
these should be `assert.equals` no?

Reply:
+1, nice catch. This is why I think we generally shouldn't use assert -- it's easy to misuse.

Assert doesn't check equality, it checks truthiness of the first argument and treats the second argument as an error message.